scipy/optimize/tests/test_trustregion_exact.py

```diff
@@ -0,0 +1,360 @@
+"""
+Unit tests for trust-region iterative subproblem.
+
+To run it in its simplest form::
+  nosetests test_optimize.py
+
+"""
+from __future__ import division, print_function, absolute_import
+
+import numpy as np
+from scipy.optimize._trustregion_exact \
+    import (estimate_smallest_singular_value,
+            singular_leading_submatrix,
+            IterativeSubproblem)
+from scipy.linalg import (svd, get_lapack_funcs, det,
+                          cho_factor, cho_solve, qr,
+                          eigvalsh, eig, norm)
+from numpy.testing import (TestCase, assert_, assert_array_equal,
+                           assert_equal, assert_array_almost_equal,
+                           assert_array_less, run_module_suite)
+
+
+def random_entry(n, min_eig, max_eig, case):
+
+    # Generate random matrix
```

Comment:
A quick note about comments. It is recommended to treat them as proper sentences https://www.python.org/dev/peps/pep-0008/#comments (including ending with a period).

For me it seems to really improve how the code feels and looks. Please don't bother to fix it here, maybe consider for future.

Also I don't really support writing too many comments, especially literally duplicating code below  (I already mentioned it in the implementation code). Again don't bother about it here.